### ISSUE TITLE
Rendering bug for map which has width/height is larger than 32768?

### ISSUE THREAD

**Issue by jeremyye10** [mapnik/mapnik#3547]
Hello,

I am trying to render a map which has width=48128 and height=24064.
For this, I increased MAX_MAPSIZE of map.hpp to large value.(default value is 16 << 10).
Rendered image is below.
![world](https://cloud.githubusercontent.com/assets/12949036/20224268/0620de52-a881-11e6-99a7-0a5f64c84517.png)

I don't know why this is happened.
After some testing, I found that the maximum allowed(?) width/height value is 32768.
Any ideas?

I am using mapnik version 3.0.12 on ubuntu 14.04.
I rendered this image using example code and data of below link.
(https://github.com/mapnik/mapnik/wiki/GettingStartedInPython)
I only changed width and height of map.

If you need more info, please let me know.

Thanks,

Jeremy

**Comment by djakk** [comment]
Hello, 

I ran into the same limitation, so I'm building giant tiles (32768* 32768) ; but I have an issue : my .png are invalid. I don't know why. (Maybe my tiles are still too large ?)
Try to open your image in Firefox : Firefox will tell you if your image is invalid.

**Comment by djakk** [comment]
Yes ! Firefox does accept a giant png tile which sizes 16384* 16384px ; so you have to build an image below mapnik limitation …

**Comment by jeremyye10** [comment]
@djakk Thanks for your comment.

Yes, you're right. Firefox cannot open this image. Additionally, I also tried to open it with Ubuntu default image viewer, IE and Chrome, and they all failed. Only windows default image viewer can open this image. A strange thing is that all the texts are correctly rendered.

**Comment by jeremyye10** [comment]
If you put some texts near by the edge of the map, you can see them correctly rendered.
